### PR TITLE
[CodeGen] Switch tiling only cases to use scf::tileUsingSCFForOp

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Comment by hanhanW** [comment]
It's based on https://github.com/openxla/iree/pull/15938

File: compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp:780]: We should probably update the logic, like [this](https://github.com/llvm/llvm-project/pull/75410/commits/c07f7e1c5c6f8bbc7189e96096004d39a0a1aa3f). We can return the cloned op, so it does not return a failure. This can be a follow-up.

https://github.com/llvm/llvm-project/blob/f4b5be1ecdc85ca4257b739afb8d57e23c7a8030/mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp#L283-L289

File: compiler/src/iree/compiler/Codegen/SPIRV/Utils.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/SPIRV/Utils.cpp]: nit: spell out type

**Review by qedawkins — APPROVED** [review]
I would wait to give Mahesh a chance to look (although trying not to bother too much during vacation).

**Comment by hanhanW** [comment]
> I would wait to give Mahesh a chance to look (although trying not to bother too much during vacation).

I think this is okay because we are towards moving out linalg::tileLinalgOp.

**Comment by qedawkins** [comment]
> > I would wait to give Mahesh a chance to look (although trying not to bother too much during vacation).
> 
> I think this is okay because we are towards moving out linalg::tileLinalgOp.

Sounds good, and by give a chance I mainly meant don't merge right away :)

**Pull request merged**
